rho/inventory_scan.py

```diff
@@ -125,12 +135,85 @@ def inventory_scan(hosts_yml_path, facts_to_collect, report_path,
 
     log_path = log_path or utilities.SCAN_LOG_PATH
 
-    print('Running:', cmd_string)
-
-    process = ansible_utils.run_with_vault(
-        cmd_string, vault_pass,
-        log_path=log_path,
-        log_to_stdout=True,
-        ansible_verbosity=verbosity)
+    try:
+        ansible_utils.run_with_vault(
+            cmd_string, vault_pass,
+            log_path=log_path,
+            log_to_stdout=True,
+            ansible_verbosity=verbosity,
+            print_before_run=True)
+    except ansible_utils.AnsibleProcessException as ex:
+        print(t("An error has occurred during the scan. Please review" +
+                " the output to resolve the given issue: %s" % str(ex)))
+        sys.exit(1)
 
-    return process.exitstatus == 0 and process.signalstatus is None
+    with open(variables_path, 'r') as variables_file:
+        vars_by_host = json.load(variables_file)
+
+    # Postprocess Ansible output
+    facts_out = []
+    for _, host_vars in utilities.iteritems(vars_by_host):
+        this_host = {}
+
+        this_host.update(host_vars['connection'])
+        this_host.update(host_vars['cpu'])
+        this_host.update(host_vars['date'])
+        this_host.update(host_vars['dmi'])
+        this_host.update(host_vars['etc_release'])
+        this_host.update(host_vars['file_contents'])
+        this_host.update(host_vars['redhat_packages'])
+        this_host.update(host_vars['redhat_release'])
+        this_host.update(host_vars['subman'])
+        this_host.update(host_vars['uname'])
+        this_host.update(host_vars['virt'])
+        this_host.update(host_vars['virt_what'])
+
+        this_host.update(
+            postprocessing.process_jboss_versions(facts_to_collect, host_vars))
+        this_host.update(
+            postprocessing.process_addon_versions(facts_to_collect, host_vars))
+        this_host.update(
+            postprocessing.process_id_u_jboss(facts_to_collect, host_vars))
+        this_host.update(
+            postprocessing.process_jboss_eap_common_files(
+                facts_to_collect, host_vars))
+        this_host.update(
+            postprocessing.process_jboss_eap_processes(
+                facts_to_collect, host_vars))
+        this_host.update(
+            postprocessing.process_jboss_eap_packages(
+                facts_to_collect, host_vars))
+        this_host.update(
+            postprocessing.process_jboss_eap_locate(
+                facts_to_collect, host_vars))
+        this_host.update(
+            postprocessing.process_jboss_eap_init_files(
+                facts_to_collect, host_vars))
+
+        postprocessing.handle_systemid(facts_to_collect, this_host)
+        postprocessing.handle_redhat_packages(facts_to_collect, this_host)
+        postprocessing.escape_characters(this_host)
```

Comment:
Seems not to be handling things correctly. Could be a change in the expected input, as I'm seeing newlines not being converted to spaces in the output.

Reply:
The input data was unicode, where as before it was strings.

I just pushed a commit with a fix. Thank you for catching this!